### PR TITLE
fix inner ref

### PR DESCRIPTION
**Pull request by mshwery**
replaces #41 

Makes `innerRef` a bit more accurate. This will need to target both v2.1.2 and v3.0.1

### CODE REVIEW

**Comment by tusbar** [comment]
Shouldn’t this also update the `innerRef` prop-types from `box.tsx` to support `RefObject`?

File: src/box.tsx [hunk 1]
```diff
@@ -21,7 +21,10 @@ Box.displayName = 'Box'
 
 Box.propTypes = {
   ...propTypes,
-  innerRef: PropTypes.func,
+  innerRef: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.shape({ current: PropTypes.elementType })
```
Comment (mshwery) [inline src/box.tsx]: @tusbar thinking something like this would work?

Comment (tusbar) [reply src/box.tsx]: Shouldn’t it be `{current: PropTypes.element}`?

Comment (mshwery) [reply src/box.tsx]: Won't that exclude other components like `MyComponent` as a ref?

Comment (tusbar) [reply src/box.tsx]: Correct me if I’m wrong but isn’t the difference between element/elementType the same between instance/class?

`<Foo />` is an `element` and `Foo` is an `elementType`?

I’ve never used `PropTypes.elementType` before but that’s what I understand from the doc https://reactjs.org/docs/typechecking-with-proptypes.html (which is not really explaining much).

Comment (tusbar) [reply src/box.tsx]: https://github.com/facebook/react/tree/master/packages/react-is#determining-an-elements-type

**Comment by ActiveObject** [comment]
Hi folks, have you been thinking about using `React.forwardRef()`? Is there any obstacle that prevents that? 

@mshwery

**Comment by mshwery** [comment]
@ActiveObject TBH we haven't really thought much about it since `innerRef` handles refs for you. Is there a use case you have that isn't currently covered by `innerRef`?

**Comment by ActiveObject** [comment]
@mshwery I had problems passing `React.createRef()` but looks like you've already covered that. Other from that `innerRef` handles all cases. 

I still think `ref` would be more intuitive for devs, as we use the same name for the rest of basic elements.

**Pull request merged**